### PR TITLE
Add tag/untag commands for container repositories

### PR DESCRIPTION
**Pull request by gerrod3**
fixes: #423

See issue for discussion around command structure. I would like to have #425 merged first so I can use those commands for creating tests for these commands. I'll update the `suppported_workflows.md` after I'm done with all the new container commands to avoid merge conflicts.

### CODE REVIEW

File: pulpcore/cli/container/context.py
Comment (gerrod3) [inline pulpcore/cli/container/context.py:83]: If adding/removing tags between the two repository types is supported from different versions of pulp_container then I can add capabilities here. Might be worth to do it anyway since I don't know which version tagging was added in.

File: pulpcore/cli/container/repository.py
Comment (gerrod3) [inline pulpcore/cli/container/repository.py:138]: Tried doing some bare minimum checks and leaving the rest to the endpoint's serializer. If there is anything else needed please tell me.

**Review by gerrod3 — COMMENTED** [review]
Commands are pretty simple, but the main question I have is about their placement. I would like for @pulp/container-plugin to be in consensus so it doesn't have to be changed later.

File: pulpcore/cli/container/context.py
Comment (lubosmj) [reply pulpcore/cli/container/context.py:83]: Support for tagging manifests within push repositories was added in version 2.3.0 (2021-02-08). According to the pulp-cli [docs](https://github.com/pulp/pulp_container/blob/2.3/requirements.txt#L1), we support `5 versions of pulpcore from 3.11-3.16 and 5 of Pulp's plugins: pulp_ansible, pulp_container`. Version 2.3.0 falls into the pulpcore requirement when looking at [requirements.txt](https://github.com/pulp/pulp_container/blob/b81d233f0ce950153cfea9165cd7b4c2aad3286f/requirements.txt#L1).

File: pulpcore/cli/container/context.py
Comment (mdellweg) [inline pulpcore/cli/container/context.py:107]: This looks like a rebase is needed.

Comment (gerrod3) [reply pulpcore/cli/container/context.py:107]: Are you talking about your new PR refactoring the IDs? #448 I can wait till after it's merged to rebase.

Comment (mdellweg) [reply pulpcore/cli/container/context.py:107]: Oh, yes, i get confused what's merged or not.

File: pulpcore/cli/container/repository.py
Comment (mdellweg) [inline pulpcore/cli/container/repository.py:46]: FWIW, an empty string would not pass the regex. But i see this check provide a better error condition.

File: pulpcore/cli/container/repository.py
Comment (mdellweg) [inline pulpcore/cli/container/repository.py:150]: (Note for later:) Ha, we should have a `needs_capability` that raises itself.
Rationale: All mentions of Pulp versions should be in `context.py`.

**Review by mdellweg — APPROVED** [review]

**Pull request merged**
